ibis/backends/base/__init__.py

```diff
@@ -40,6 +41,57 @@ def connect(connection_string, **options):
         Connect to the underlying database and return a client object.
         """
 
+    # @abc.abstractmethod
```

Comment:
is the intent to uncomment?

Reply:
Yes. I can't make them abstract in this PR, since the methods are not implemented in the subclasses yet (to not make this PR huge). Immediately after this is merged, I'll open a PR for each of these methods, with the same approach I did for `version` here. I'll uncomment in those, so didn't bother to add a comment, since it won't stay long commented.